### PR TITLE
Prepend changes instead of Appending to Changelog

### PR DESCRIPTION
**Pull request by bjoernh2000**
## This introduces a breaking change

- [ ] Yes
- [ ] No

## This Pull Request implements

Issue#736 - prepending instead of appending to CHANGELOG.md

## Description

Made Prepending instead of appending, edited README in versioning to match current state, fixed some small bugs that would cause errors in edge cases

### CODE REVIEW

**Review by KPostOffice — APPROVED** [review]
/lgtm

Can you point me to your

File: kebechet/managers/version/version.py [hunk 1]
```diff
@@ -299,14 +301,15 @@ def _compute_changelog(
             ).splitlines()
 
         if version_file:
-            # TODO: We should prepend changes instead of appending them.
             _LOGGER.info("Adding changelog to the CHANGELOG.md file")
-            with open("CHANGELOG.md", "a") as changelog_file:
+            with open("CHANGELOG.md", "r+") as changelog_file:
+                content = changelog_file.read()
+                changelog_file.seek(0, 0)
```
Comment (fridex) [inline kebechet/managers/version/version.py:307]: Will this work with the title that is present in CHANGELOG file?

Comment (bjoernh2000) [reply kebechet/managers/version/version.py:307]: Oh I see I don't think it'll work with a title, it'll prepend to the front of the file

File: kebechet/managers/version/version.py [hunk 2]
```diff
@@ -303,13 +303,33 @@ def _compute_changelog(
...
-                content = changelog_file.read()
+                lines = changelog_file.readlines()
                 changelog_file.seek(0, 0)
-                changelog_file.write(
-                    f"\n## Release {new_version} ({datetime.now().replace(microsecond=0).isoformat()})\n"
-                )
-                changelog_file.write("\n".join(changelog))
-                changelog_file.write("\n" + content)
+                if (
+                    lines[0][0] == "#" and lines[0][1] == " "
```
Comment (fridex) [inline kebechet/managers/version/version.py:309]:
```suggestion
                    lines[0].startswith("# ")
```

**Review by fridex — COMMENTED** [review]
lgtm, one minor change and this can go in 👍🏻 thanks!

**Review by fridex — APPROVED** [review]
lgtm, thanks! 👍🏻

**Review by fridex — APPROVED** [review]

**Pull request merged**